batch_consumer.go

```diff
@@ -47,9 +49,11 @@ func (b *batchConsumer) GetMetric() *ConsumerMetric {
 
 func (b *batchConsumer) Consume() {
 	go b.subprocesses.Start()
+
 	b.wg.Add(1)
 	go b.startConsume()
 
+	b.wg.Add(b.concurrency)
```

Comment:
For more readability, I extract wg.Add within setupConcurrentWorkers method here